core/trino-main/src/main/java/io/trino/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -135,6 +143,82 @@ public static PlanNodeStatsEstimate capStats(PlanNodeStatsEstimate stats, PlanNo
         return result.build();
     }
 
```

Comment:
We apply it for every predicate. It would probably be better to have
```
intersectCorrelatedStats(Iterable<PlanNodeStatsEstimate> stats)
```
because otherwise this might have a performance impact (e.g. we construct `map` with every column each time), especially in join. Note that in `io.trino.cost.PlanNodeStatsEstimate` we use `PMap` to speed-up modifications of single column.